src/main/java/com/fakng/fakngagrgtr/parser/google/GoogleParser.java

```diff
@@ -15,30 +12,28 @@
 import javax.annotation.PostConstruct;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Optional;
 
 @Component
 public class GoogleParser extends ApiParser {
 
-    private static final String GOOGLE_NAME = "Google";
-    private Company google;
-
-    public GoogleParser(WebClient webClient, LocationCache locationCache, CompanyRepository companyRepository,
-                        LocationRepository locationRepository, @Value("${url.google}") String url) {
-        super(webClient, locationCache, companyRepository, locationRepository);
+    public GoogleParser(WebClient webClient, CompanyRepository companyRepository, LocationProcessor locationProcessor,
+                        @Value("${url.google}") String url) {
+        super(webClient, companyRepository, locationProcessor);
         this.url = url;
     }
 
     @PostConstruct
     public void init() {
-        google = companyRepository.findByTitle(GOOGLE_NAME)
-                .orElseThrow(() -> new IllegalStateException(String.format("There is no %s company present in DB", GOOGLE_NAME)));
-        google.getLocations()
-                .forEach(location -> locationCache.putIfAbsent(location.getCity(), location.getCountry(), location));
+        initBase();
     }
 
     @Override
-    protected List<Vacancy> getAllVacancies() throws Exception {
+    protected String getCompanyName() {
+        return "Google";
```

Comment:
Where this method used?
May be declare as constant

Reply:
It's used in the base class. No, it'll make it obligatory for everyone to consistently specify company names. I think it's better than having their own constant variables in all classes